packages/backend/src/api/controllers/status/StatusController.ts

```diff
@@ -23,31 +18,22 @@ export class StatusController {
   ) {}
 
   async getPricesStatus() {
-    const boundaries = await this.priceRepository.calcDataBoundaries()
-
-    const result: PriceStatus[] = []
-
-    for (const token of this.tokens) {
-      const boundary = boundaries.get(token.id)
-
-      if (!boundary) {
-        result.push({
-          coingeckoId: token.coingeckoId,
-          min: '',
-          max: '',
-          message: '!!! no prices !!!',
-        })
-        continue
-      }
-
-      result.push({
-        coingeckoId: token.coingeckoId,
-        min: unixTimeToString(boundary.earliest),
-        max: unixTimeToString(boundary.latest),
-        message: getSyncStatus(boundary.latest),
+    const latestByToken = await this.priceRepository.getLatestByToken()
+
+    const statuses = this.tokens
+      .map((token): Status => {
+        const latest = latestByToken.get(token.id)
+
+        return {
+          name: token.coingeckoId.toString(),
+          timestamp: latest?.timestamp.toNumber(),
+          value: latest?.priceUsd.toString(),
+          isSynced: isSynced(latest?.timestamp),
+        }
       })
-    }
-    return result
+      .sort((a, b) => Number(a.isSynced) - Number(b.isSynced))
```

Comment:
Wouldn't it be nicer to sort by `timestamp` itself? That way it's guaranteed that two rows with the same value of `isSynced` are actually ordered by what was the list time they were synced.

Reply:
this feature was intended to move unsynced values to the top, by sorting with timestamp we will loose the alphabetical order. I am not sure whether this is intended behaviour.